frontend/src/modules/editor/contentObject/views/editorMenuView.js

```diff
@@ -39,38 +40,35 @@ define(function(require){
     * Renders all menu layers from the current course to the Origin.editor.currentContentObject
     */
     renderLayers: function() {
-      var selectedModel = Origin.editor.currentContentObject;
-      // no previous state, so should only render the first level
-      if(!selectedModel) {
-        this.renderLayer(Origin.editor.data.course);
-        return;
-      }
-      // check if we can reuse any existing layers, and only render the new ones
+      var selectedModel = Origin.editor.currentContentObject || Origin.editor.data.course;
       this.getItemHeirarchy(selectedModel, function(hierarchy) {
-        var index;
-        var renderedLayers = this.$('.editor-menu-layer');
-        for(var i = 0, count = hierarchy.length; i < count; i++) {
-          if($(renderedLayers[i]).attr('data-parentid') === hierarchy[i].get('_id')) {
-            index = i+1;
+        var ids = [];
+
+        for (var i = 0; i < hierarchy.length; i++) {
+          var item = hierarchy[i];
+          var id = item.get('_id');
+          ids.push(id);
+          if (!this.layerViews.hasOwnProperty(id)) {
+            this.renderLayer(item);
           }
         }
-        // we can reuse layers up to 'index', remove the rest
-        if(index !== undefined) {
-          hierarchy = hierarchy.slice(index);
-          var layersToRemove = renderedLayers.slice(index);
-          for(var i = 0, count = layersToRemove.length; i < count; i++) {
-            layersToRemove[i].remove();
+
+        // remove all unused layerviews 
+        for (var id in this.layerViews) {
```

Comment:
Can this be refactored to be a bit more readable? Something like:
```
// remove all unused layerviews 
for (var id in this.layerViews) {
  if (!this.layerViews.hasOwnProperty(id) || ids.indexOf(id) > -1) {
    continue;
  }
  this.layerViews[id].remove();
  delete this.layerViews[id];
}
```
*think* that does the same...

Reply:
thanks, fixed it

Reply:
Yeah, think the loop was fine, just the if condition

Reply:
updated